rho/utilities.py

```diff
@@ -104,42 +131,46 @@ def tail_discovery_scan(path, ansible_verbosity):
     :param path: tuple containing the path to file to follow
     :param ansible_verbosity: the verbosity level
     """
-    hosts_processed = 0
-    hosts_successful = 0
-    hosts_unreachable = 0
-    hosts_failed = 0
     if len(path) > 0:  # pylint: disable=len-as-condition
         # pylint: disable=no-member
-        for line in sh.tail('-f', '-n', '+0', path, _iter=True):
-            print_status = False
-            line = line.strip('\n')
-            if 'SUCCESS' in line:
-                hosts_successful += 1
-                hosts_processed += 1
-                print_status = True
-            elif 'FAILED' in line:
-                hosts_failed += 1
-                hosts_processed += 1
-                print_status = True
-            elif 'UNREACHABLE' in line:
-                hosts_unreachable += 1
-                hosts_processed += 1
-                print_status = True
-
-            # Display every 5 processed
-            if hosts_processed % 5 == 0 and print_status:
-                print(_('%d hosts processed with the current credential. ' %
-                        hosts_processed))
-                if hosts_successful > 0:
-                    print(_('%d hosts connected successfully with '
-                            'the current credential.' % hosts_successful))
-                if hosts_failed > 0:
-                    print(_('%d hosts failed to connect with the '
-                            'current credential.' % hosts_failed))
-                if hosts_unreachable > 0:
-                    print(_('%d hosts were unreachable.' % hosts_unreachable))
-
-                print('\n')
+        process = sh.tail('-f', '-n', '+0', path, _out=process_discovery_scan,
+                          _bg=True, _bg_exc=False)
+        return process
+
+
+def process_host_scan(line, stdin_queue, process):
+    """Process the output of a discovery scan.
+
+    :param line: a line from the discovery scan_log
+    """
+    ansible_verbosity = int(os.environ.get('ANSIBLE_VERBOSITY', '1'))
+    truncate = 1
+    if ansible_verbosity:
+        truncate = ansible_verbosity
+
+    print_line = truncate
+    plabook_started = False
+    truncated = False
+    line = line.strip('\n')
+    if line.startswith('TASK') or line.startswith('PLAY'):
+        print(line)
+        print_line = truncate
+        plabook_started = True
+        truncated = False
+    elif 'FAILED!' in line or 'module_stderr' in line:
+        print(line)
+        print_line = truncate if truncate > 3 else 4
+    elif print_line > 0:
+        line_len = len(line)
+        char_truncate = truncate * 100
+        if line_len > char_truncate:
+            print(line[0:char_truncate] + '...')
+        else:
+            print(line)
+        print_line = print_line - 1
+    elif print_line == 0 and not truncated and plabook_started:
+        print(_('-- output truncated --'))
+        truncated = True
 
 
 def tail_host_scan(path, ansible_verbosity):
```

Comment:
Have you considered combining `tail_discovery_scan` and `tail_host_scan` into one function with a parameter for the function to tail with? I don't feel strongly about this, but it might clean up the code a little bit.